### PR TITLE
Running muon and neutron data in fit benchmarking

### LINKED ISSUE #20865: Fit benchmarking adding Muon data
This is to allow the fit benchmarking tool to fit to muon data (outside of neutron file). @Anders-Markvardsen is there a good place to update the fitting data (i.e. add the new muon data)?

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The fit bench marking should be able to use neutron and muon data. This required adding the full path to the load algorithm and making muon an acceptable argument.

There are few other changes to allow fit bench marking work with the new output from the fit algorithm. 

**To test:**

Download the attached zip file: 

[FitMinimizerTesting.zip](https://github.com/mantidproject/mantid/files/1379578/FitMinimizerTesting.zip)

Unzip it

In the python scripting window open runScripts.py

Then on line 35 (run_data) you can choose what to run (muon, neutron or nist). Run each one in turn. 

Plots and summary tables should be produced in the mantidplot folder of your build. There will be 2 plots (start and best fit) for each test. 

Fixes #20865 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/CompareFitMinimizers/fitting_benchmarking.py [hunk 1]
```diff
@@ -336,38 +346,36 @@ def run_fit(wks, prob, function, minimizer='Levenberg-Marquardt', cost_function=
...
         # Note the ugly adhoc exception. We need to reconsider these WISH problems:
         if 'WISH17701' in prob.name:
             ignore_invalid = False
-        status, chi2, covar_tbl, param_tbl, fit_wks = msapi.Fit(function, wks, Output='ws_fitting_test',
-                                                                Minimizer=minimizer,
-                                                                CostFunction=cost_function,
-                                                                IgnoreInvalidData=ignore_invalid,
-                                                                StartX=prob.start_x, EndX=prob.end_x)
+
+        status = msapi.Fit(function, wks, Output='ws_fitting_test',
```
Comment (Anders-Markvardsen) [inline scripts/CompareFitMinimizers/fitting_benchmarking.py]: I think I agree better with capture result from fit in one variable rather than individually. But if you do this do perhaps the return variable should be result or fit_result rather than status?

File: scripts/CompareFitMinimizers/fitting_benchmarking.py [hunk 2]
```diff
@@ -336,38 +346,36 @@ def run_fit(wks, prob, function, minimizer='Levenberg-Marquardt', cost_function=
...
-    chi2 = None
     param_tbl = None
-    fit_wks = None
     try:
         # When using 'Least squares' (weighted by errors), ignore nans and zero errors, but don't
         # ignore them when using 'Unweighted least squares' as that would ignore all values!
         ignore_invalid = cost_function == 'Least squares'
-
         # Note the ugly adhoc exception. We need to reconsider these WISH problems:
         if 'WISH17701' in prob.name:
```
Comment (Anders-Markvardsen) [inline scripts/CompareFitMinimizers/fitting_benchmarking.py]: This exception deserve a separate pull request to get fixed....

File: scripts/CompareFitMinimizers/fitting_benchmarking.py (hunk 1 above)
Comment (AnthonyLim23) [reply scripts/CompareFitMinimizers/fitting_benchmarking.py]: yeah, I can change the variable name. I just picked status as it was the name I found in an example

File: scripts/CompareFitMinimizers/fitting_benchmarking.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/CompareFitMinimizers/fitting_benchmarking.py]: sorry, which exception? The WISH one?

**Comment by Anders-Markvardsen** [comment]
See below results from running muon examples with a couple of minimizers - which demo that this pull requests works and as a side an interesting result

capture

**Review by Anders-Markvardsen — CHANGES_REQUESTED** [review]
Looks good. I just have one minor suggestion for change for you do or argue against first

**Comment by AnthonyLim23** [comment]
if it is just the variable name, I will change that first thing tomorrow

File: scripts/CompareFitMinimizers/fitting_benchmarking.py (hunk 2 above)
Comment (Anders-Markvardsen) [reply scripts/CompareFitMinimizers/fitting_benchmarking.py]: Yes just WISH17701:)

**Comment by Anders-Markvardsen** [comment]
Yes just the variable name

**Review by Anders-Markvardsen — APPROVED** [review]

**Comment by martyngigg** [comment]
There appear to be `flake8` warnings here. Can we fix these, please?

**Comment by AnthonyLim23** [comment]
fixed the flake8 warnings

**Comment by ianbush** [comment]
Do the release notes need updating?

**Review by ianbush — CHANGES_REQUESTED** [review]
Please either add the release notes, or state if it does not need to be in them.

**Comment by AnthonyLim23** [comment]
it does not need to be in the release notes. This is more of a diagnostic tool for us

**Review by ianbush — APPROVED** [review]

**Comment by ianbush** [comment]
Thanks for clarifying @AnthonyLim23 . When you open a PR please use the template to state this in the description, else it is not clear if it has been forgotten or was not intended.
